src/middlewares/verifyUserAccess.js

```diff
@@ -0,0 +1,29 @@
+const userSchema = require('../model/user.model')
+const jwt = require('jsonwebtoken')
+
+const veridyAccess = (req, res, next) => {
+  try{
+    let token = req.headers.authorization
+    let decode = jwt.decode(token, process.env.SECRET)
+    const id = req.params.userId
+
+    userSchema.findOne({email: decode.data.email}, (err, user) => {
+      try{
+        if(user.role === 'admin') {
+          console.log('eres admin crack');
+          next()
+        }if(user._id == id){
+          next()
+        }else{
+          res.status(403).send({msg:'You can´t edit another account if it´s not yours'})
+        }
+      }catch(error){
+        res.status(401).send({error: error.message})
+      }
+    })
+  }catch(error){
+    res.status(401).send({ error: error.message })
+  }
```

Comment:
- Is shouldn't be a lot a res.status, instead of that you can return an error and the catch will handle it
- Change the console.log message when is an admin
- You wrote **veridyAccess** instead of **verifyAccess**